main.tf

```
@@ -1 +1,200 @@
+  )
+  iam_role_arn         = module.lacework_ct_iam_role.created ? module.lacework_ct_iam_role.arn : var.iam_role_arn
+  iam_role_external_id = module.lacework_ct_iam_role.created ? module.lacework_ct_iam_role.external_id : var.iam_role_external_id
+  iam_role_name = var.use_existing_iam_role ? var.iam_role_name : (
+    length(var.iam_role_name) > 0 ? var.iam_role_name : "${var.prefix}-iam-${random_id.uniq.hex}"
+  )
+}
+
+resource "random_id" "uniq" {
+  byte_length = 4
+}
+
+
+resource "aws_sqs_queue" "lacework_cloudtrail_sqs_queue" {
+  provider = aws.audit
+  name = local.sqs_queue_name
+  tags = var.tags
+}
+
+resource "aws_sqs_queue_policy" "lacework_sqs_queue_policy" {
+  provider = aws.audit
+  depends_on = [time_sleep.wait_time]
+  queue_url = aws_sqs_queue.lacework_cloudtrail_sqs_queue.id
+  policy    = <<POLICY
+{
+	"Version": "2012-10-17",
+	"Id": "lacework_sqs_policy_${random_id.uniq.hex}",
+	"Statement": [
+		{
+			"Sid": "AllowLaceworkSNSTopicToSendMessage",
+			"Effect": "Allow",
+			"Principal": {
+				"AWS": "*"
+			},
+			"Action": "SQS:SendMessage",
+			"Resource": "${aws_sqs_queue.lacework_cloudtrail_sqs_queue.arn}",
+			"Condition": {
+				"ArnEquals": {
+					"aws:SourceArn": "${local.sns_topic_arn}"
+				}
+			}
+		},
+		{
+			"Sid": "AllowLaceworkCrossAccountRoleToAccessSQSQueue",
+			"Effect": "Allow",
+			"Principal": {
+				"AWS": "${local.iam_role_arn}"
+			},
+			"Action": [
+				"sqs:DeleteMessage",
+				"sqs:ReceiveMessage"
+			      ],
+			"Resource": "${aws_sqs_queue.lacework_cloudtrail_sqs_queue.arn}"
+		}
+	]
+}
+POLICY
+}
+
+resource "aws_sns_topic_subscription" "lacework_sns_topic_sub" {
+  provider = aws.audit
+  topic_arn = local.sns_topic_arn
+  protocol  = "sqs"
+  endpoint  = aws_sqs_queue.lacework_cloudtrail_sqs_queue.arn
+}
+
+data "aws_iam_policy_document" "cross_account_policy" {
+  provider = aws.log_archive
+  version = "2012-10-17"
+  
+  statement {
+    sid       = "ReadLogFiles"
+    actions   = ["s3:Get*"]
+    resources = ["arn:aws:s3:::aws-controltower-logs-${var.aws_log_archive_account_id}-${var.control_tower_region}/${var.aws_organization_id}/*AWSLogs/*"]
+  }
+
+  statement {
+    sid       = "GetAccountAlias"
+    actions   = ["iam:ListAccountAliases"]
+    resources = ["*"]
+  }
+
+  statement {
+    sid       = "ListLogFiles"
+    resources = ["arn:aws:s3:::aws-controltower-logs-${var.aws_log_archive_account_id}-${var.control_tower_region}/${var.aws_organization_id}/*AWSLogs/*"]
+    actions   = ["s3:ListBucket"]
+
+    condition {
+      test     = "StringLike"
+      variable = "s3:prefix"
+      values   = ["*AWSLogs/"]
+    }
+  }
+
+  statement {
+    sid       = "ConsumeNotifications"
+    # resources = ["${aws_sqs_queue.lacework_cloudtrail_sqs_queue.arn}"]
+    resources = flatten([var.sqs_queues, [aws_sqs_queue.lacework_cloudtrail_sqs_queue.arn]])
```

You do not need this flattening since we used it a year and a half ago when there was no
concept of Org Account Mappings
```suggestion
    resources = ["${aws_sqs_queue.lacework_cloudtrail_sqs_queue.arn}"]
```